pallets/pools/src/lib.rs

```diff
@@ -444,18 +441,14 @@ pub mod pallet {
 			Self::is_valid_tranche_change(&Vec::new(), &tranches)?;
 
 			let now = T::Time::now().as_secs();
-			let n_tranches = tranches.clone().len() as u32;
 			let tranches = tranches
 				.into_iter()
 				.enumerate()
 				.map(|(tranche_id, tranche)| Tranche {
 					interest_per_sec: tranche.interest_per_sec.unwrap_or(One::one()),
 					min_risk_buffer: tranche.min_risk_buffer.unwrap_or(Perquintill::zero()),
-					seniority: tranche.seniority.unwrap_or(
-						n_tranches
-							.saturating_sub(1)
-							.saturating_sub(tranche_id as u32),
-					),
+					// seniority increases as index since the order is from junior to senior
+					seniority: (tranche_id as u32),
```

Comment:
Should still respect the manually specified seniority if it's specified?

Reply:
Yeah it should take `tranche.seniority` unless it's `None`.

Reply:
Is seniority not incremental like the index? What do seniority here means @offerijns ?

Reply:
By default, seniority will be the tranche id, meaning it will be 0 for junior, 1 for the senior, in a 2 tranche structure. This is used for the submission scoring weights: a higher seniority tranche should have higher weights, to give the tranches preference in epoch execution. However, I purposefully already included the ability to override the seniority property, to support tranches with the same seniority. As in, it will be useful (at some point) to make it possible to have multiple tranches with the same seniority.

Reply:
can we bring that feature of allowing custom seniority down the line? It will be a runtime change and should not touch the storage ?